protocols/raft/src/main/java/io/atomix/protocols/raft/storage/snapshot/impl/DefaultSnapshot.java

```diff
@@ -159,6 +160,14 @@ public void closeWriter(final SnapshotWriter writer) {
     this.writer = null;
   }
 
+  @Override
+  public int compareTo(final Snapshot other) {
+    return Comparator.comparingLong(Snapshot::index)
+        .thenComparingLong(Snapshot::term)
+        .thenComparing(Snapshot::timestamp)
+        .compare(this, other);
```

Comment:
Would like to discuss this - actually Idk what will happen here

Reply:
Oops, as this is in the default then we dont need to overload here